addons/mail/static/src/models/user/user.js

```diff
@@ -113,7 +113,7 @@ function factory(dependencies) {
                 thread.model === 'mail.channel' &&
                 thread.public === 'private'
             );
-            if (!chat) {
+            if (!chat ||!chat.isPinned) {
```

Comment:
Don't like this. We have `chat` locally, so no reason to fetch info from server. Not sure, but I guess is to have side-effect to make channel pin? Would just focus on making it pinned, and explicitly so.

Reply:
Is there any possibility that the chat has been updated in a way it's not updated locally so that the fetch makes sense ?

@seb-odoo what do you think of the side-effect proposed by @alexkuhn instead of the fetch ?

Reply:
Python should be cleaned up, but as it stands, after checking I think both are bad.

The only difference between the 2 methods in the current scenario is that `channel_get` is returning the `channel_info` whereas `channel_pin` is doing absolutely nothing (in terms of return value).

`channel_get` is also sending on the bus but only when a new channel is created, not when re-pinning an existing one. This makes no sense.
And `channel_pin` is also sending on the bus, but only when unpinning, which also makes no sense.

Probably both methods should be fixed to send on the bus, always. And in that case the become almost equivalent for our current purpose. `channel_get` makes a more complex query but ensure we get the proper chat for sure, whereas using `channel_pin` is assuming the JS knows the correct channel and just wants to pin it.

Reply:
I find this a little bit strange for `channel_get` to write on the bus what the method is actually returning. And it only write on bus for others partners, not current user so for now current user never received anything from this method called by him and if it's called by someone else he already get the notification.

So for me, we only need to change `channel_pin`'s notifications if we want to use it here instead of `channel_get`

Reply:
The general rule is: method return = tab specific, bus = cross tab/browser sync.
The question to ask is whether the change should be reflected in only one tab or in all of them.

It seems natural to me that joining a channel with `channel_get`, the channel should be displayed in all tabs, so using the bus, even for current partner. Fortunately this is already sync because other RPC that are done in this flux do send on the bus (eg. `channel_fold`), but I think it would be cleaner if `channel_get` was doing it immediately.

Reply:
Ok I understand, looking into it

Reply:
Done, tell me what do you think about this. And do we leave js like that or go to channel_pin usage in user.js ?

Reply:
If we consider the server the source of truth (and we should) then keeping `channel_get` as done currently is better for reasons I explained above.

Reply:
I let @alexkuhn resolve this if he agree with you or have more arguments to oppose to that (I'm fine either way)

Reply:
I haven't followed all the technical aspects mentioned in this thread.
Not really about agreeing or opposing with what @seb-odoo said, TBH I'm satisfied that there's some thought process about this LOC.

Considering that this is an unfortunate necessity with regard to current implementation of python `/mail` methods and cross-tab longpolling, I would just put a comment that motivates the `!isPinned` in condition.